### ISSUE TITLE
setup fixture failing in agent singlebigip tests

### ISSUE THREAD

**Issue by ssorenso** [F5Networks/f5-openstack-agent#844]
* Title: setup fixture failing in agent singlebigip tests
* Attachments: 
* Details: 

```
    @pytest.fixture
    def setup(request, mgmt_root):
        def teardown():
            system_helper.delete_folder(mgmt_root, default_partition)
        request.addfinalizer(teardown)

        system_helper.create_folder(mgmt_root, {'name': default_partition,
>                                           'subPath': '/'})

test_network.py:76:
```
##### md5sums:

1fdc6bea5f01677fa7e692b7add5d365

#### OpenStack Release
* Newton

#### Agent Version
* stable/newton - `d884f8e70e84a07d5d65161df71c270cc847aefa`

#### Operating System
Nightly

#### Deployment
Tests impacted:
* openstack_agent_newton_12.1.1-overcloud.64.consoleText.TestRouteDomain.test_route_domain_get_vlans_by_id

**Comment by ssorenso** [comment]
```
    @pytest.fixture
    def setup(request, mgmt_root):
        def teardown():
            system_helper.delete_folder(mgmt_root, default_partition)
        request.addfinalizer(teardown)

        system_helper.create_folder(mgmt_root, {'name': default_partition,
>                                           'subPath': '/'})

test_network.py:76:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py:35: in create_folder
    f.create(**folder)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:974: in create
    return self._create(**kwargs)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:941: in _create
    response = session.post(_create_uri, json=kwargs, **requests_params)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <icontrol.session.iControlRESTSession object at 0x7fe642ad3a90>
RIC_base_uri = 'https://10.190.24.172:443/mgmt/tm/sys/folder/'
kwargs = {'json': {'name': 'test', 'subPath': '/'}}, partition = ''
sub_path = '', suffix = '', identifier = '', uri_as_parts = False
transform_name = False
REST_uri = 'https://10.190.24.172:443/mgmt/tm/sys/folder/'

    @functools.wraps(method)
    def wrapper(self, RIC_base_uri, **kwargs):
        partition = kwargs.pop('partition', '')
        sub_path = kwargs.pop('subPath', '')
        suffix = kwargs.pop('suffix', '')
        identifier, kwargs = _unique_resource_identifier_from_kwargs(**kwargs)
        uri_as_parts = kwargs.pop('uri_as_parts', False)
        transform_name = kwargs.pop('transform_name', False)
        if uri_as_parts:
            REST_uri = generate_bigip_uri(RIC_base_uri, partition, identifier,
                                          sub_path, suffix,
                                          transform_name=transform_name,
                                          **kwargs)
        else:
            REST_uri = RIC_base_uri
        pre_message = "%s WITH uri: %s AND suffix: %s AND kwargs: %s" %\
            (method.__name__, REST_uri, suffix, kwargs)
        logging.debug(pre_message)
        response = method(self, REST_uri, **kwargs)
        post_message =\
            "RESPONSE::STATUS: %s Content-Type: %s Content-Encoding:"\
            " %s\nText: %r" % (response.status_code,
                               response.headers.get('Content-Type', None),
                               response.headers.get('Content-Encoding', None),
                               response.text)
        logging.debug(post_message)
        if response.status_code not in range(200, 207):
            error_message = '%s Unexpected Error: %s for uri: %s\nText: %r' %\
                            (response.status_code,
                             response.reason,
                             response.url,
                             response.text)
>           raise iControlUnexpectedHTTPError(error_message, response=response)
E           iControlUnexpectedHTTPError: 409 Unexpected Error: Conflict for uri: https://10.190.24.172:443/mgmt/tm/sys/folder/
E           Text: u'{"code":409,"message":"01020037:3: The requested folder (/test) already exists.","errorStack":[],"apiError":3}'
```
Tests:
* `openstack_agent_mitaka_12.1.1-overcloud.87.consoleText.test_get_gre_tunnel_key`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_get_gre_tunnel_key`
sums:
705cd0445d8e57718d3ddc93bf314b1a

**Comment by ssorenso** [comment]
Additional hits:
```
request = <SubRequest 'setup' for <Function 'test_get_vxlan_tunnel_key'>>
mgmt_root = <f5.bigip.ManagementRoot object at 0x7fb381916a10>

    @pytest.fixture
    def setup(request, mgmt_root):
        def teardown():
            system_helper.delete_folder(mgmt_root, default_partition)
        request.addfinalizer(teardown)

        system_helper.create_folder(mgmt_root, {'name': default_partition,
>                                           'subPath': '/'})

test_network.py:76:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py:35: in create_folder
    f.create(**folder)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:974: in create
    return self._create(**kwargs)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:941: in _create
    response = session.post(_create_uri, json=kwargs, **requests_params)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <icontrol.session.iControlRESTSession object at 0x7fb3819167d0>
RIC_base_uri = 'https://10.190.24.20:443/mgmt/tm/sys/folder/'
kwargs = {'json': {'name': 'test', 'subPath': '/'}}, partition = ''
sub_path = '', suffix = '', identifier = '', uri_as_parts = False
transform_name = False
REST_uri = 'https://10.190.24.20:443/mgmt/tm/sys/folder/'

    @functools.wraps(method)
    def wrapper(self, RIC_base_uri, **kwargs):
        partition = kwargs.pop('partition', '')
        sub_path = kwargs.pop('subPath', '')
        suffix = kwargs.pop('suffix', '')
        identifier, kwargs = _unique_resource_identifier_from_kwargs(**kwargs)
        uri_as_parts = kwargs.pop('uri_as_parts', False)
        transform_name = kwargs.pop('transform_name', False)
        if uri_as_parts:
            REST_uri = generate_bigip_uri(RIC_base_uri, partition, identifier,
                                          sub_path, suffix,
                                          transform_name=transform_name,
                                          **kwargs)
        else:
            REST_uri = RIC_base_uri
        pre_message = "%s WITH uri: %s AND suffix: %s AND kwargs: %s" %\
            (method.__name__, REST_uri, suffix, kwargs)
        logging.debug(pre_message)
        response = method(self, REST_uri, **kwargs)
        post_message =\
            "RESPONSE::STATUS: %s Content-Type: %s Content-Encoding:"\
            " %s\nText: %r" % (response.status_code,
                               response.headers.get('Content-Type', None),
                               response.headers.get('Content-Encoding', None),
                               response.text)
        logging.debug(post_message)
        if response.status_code not in range(200, 207):
            error_message = '%s Unexpected Error: %s for uri: %s\nText: %r' %\
                            (response.status_code,
                             response.reason,
                             response.url,
                             response.text)
>           raise iControlUnexpectedHTTPError(error_message, response=response)
E           iControlUnexpectedHTTPError: 409 Unexpected Error: Conflict for uri: https://10.190.24.20:443/mgmt/tm/sys/folder/
E           Text: u'{"code":409,"message":"01020037:3: The requested folder (/test) already exists.","errorStack":[],"apiError":3}'

/usr/local/lib/python2.7/dist-packages/icontrol/session.py:272: iControlUnexpectedHTTPError
```
Tests:
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_get_vxlan_tunnel_key`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_get_vlan_id`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_get_selfip_addr`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_get_selfips`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_delete_selfip`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.TestRouteDomain.test_route_domain_CRUD`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.TestRouteDomain.test_route_domain_get_ids`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.TestRouteDomain.test_route_domain_get_names`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.TestRouteDomain.test_route_domain_get_vlans_by_id`
* `openstack_agent_newton_12.1.1-overcloud.64.consoleText.test_arp_delete_by_network`
Sums:
* `md5sum:b28ed8b28ed1af15410cd366f19863ae`
* `reduced(md5sum):d41d8cd98f00b204e9800998ecf8427e`

**Comment by ssorenso** [comment]
Additional sum for this issue:
00cdb4d2e38a5e5a31b7b19fb0624d39
Test:
openstack_agent_liberty_12.1.1-overcloud.92.consoleText | errors | test_delete_selfip

**Comment by ssorenso** [comment]
additional sums from today:
md5sum:dd88c734120f49d93a40a1ce58e7f288
* `openstack_agent_liberty_12.1.2-overcloud.5.consoleText`
  * Type: failures
  * Test: `test_get_mgmt_addr_by_device`
* `openstack_agent_newton_12.1.2-overcloud.5.consoleText`
  * Type: failures
  * Test: `test_featureoff_withsegid_listener`

**Comment by ssorenso** [comment]
```
mgmt_root = <f5.bigip.ManagementRoot object at 0x7f3b6f16df10>

    def test_auto_sync(mgmt_root):
        # assume BIG-IP has device group test-group
        device_group = 'test-group'
>       cm.disable_auto_sync(device_group, mgmt_root)

test_cluster.py:54:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/cluster_manager.py:36: in disable_auto_sync
    name=device_group_name, partition=partition)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1026: in load
    return self._load(**kwargs)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1005: in _load
    response = refresh_session.get(base_uri, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <icontrol.session.iControlRESTSession object at 0x7f3b6f16d790>
RIC_base_uri = 'https://10.190.25.98:443/mgmt/tm/cm/device-group/', kwargs = {}
partition = 'Common', sub_path = '', suffix = '', identifier = 'test-group'
uri_as_parts = True, transform_name = False
REST_uri = 'https://10.190.25.98:443/mgmt/tm/cm/device-group/~Common~test-group'

    @functools.wraps(method)
    def wrapper(self, RIC_base_uri, **kwargs):
        partition = kwargs.pop('partition', '')
        sub_path = kwargs.pop('subPath', '')
        suffix = kwargs.pop('suffix', '')
        identifier, kwargs = _unique_resource_identifier_from_kwargs(**kwargs)
        uri_as_parts = kwargs.pop('uri_as_parts', False)
        transform_name = kwargs.pop('transform_name', False)
        if uri_as_parts:
            REST_uri = generate_bigip_uri(RIC_base_uri, partition, identifier,
                                          sub_path, suffix,
                                          transform_name=transform_name,
                                          **kwargs)
        else:
            REST_uri = RIC_base_uri
        pre_message = "%s WITH uri: %s AND suffix: %s AND kwargs: %s" %\
            (method.__name__, REST_uri, suffix, kwargs)
        logging.debug(pre_message)
        response = method(self, REST_uri, **kwargs)
        post_message =\
            "RESPONSE::STATUS: %s Content-Type: %s Content-Encoding:"\
            " %s\nText: %r" % (response.status_code,
                               response.headers.get('Content-Type', None),
                               response.headers.get('Content-Encoding', None),
                               response.text)
        logging.debug(post_message)
        if response.status_code not in range(200, 207):
            error_message = '%s Unexpected Error: %s for uri: %s\nText: %r' %\
                            (response.status_code,
                             response.reason,
                             response.url,
                             response.text)
>           raise iControlUnexpectedHTTPError(error_message, response=response)
E           iControlUnexpectedHTTPError: 404 Unexpected Error: Not Found for uri: https://10.190.25.98:443/mgmt/tm/cm/device-group/~Common~test-group
E           Text: u'{"code":404,"message":"01020036:3: The requested device group (/Common/test-group) was not found.","errorStack":[],"apiError":3}'

/usr/local/lib/python2.7/dist-packages/icontrol/session.py:272: iControlUnexpectedHTTPError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_auto_sync`
  * Sum: `md5sum:bbe60c417fd669badd7e6b8169fd9612`

**Comment by richbrowne** [comment]
Fixed by: 0cdebc5be80dc4ba8759672f6ee808123c590e5e
